curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md

```diff
@@ -28,27 +28,31 @@ Find the maximum total from top to bottom in `numTriangle`, a 2D array defined i
 `maximumPathSumII(testTriangle)` should return a number.
 
 ```js
-assert(typeof maximumPathSumII(testTriangle) === 'number');
+assert(typeof maximumPathSumII(_testTriangle) === 'number');
```

Comment:
Why the change in variable names?

Reply:
I wanted to avoid any possible conflict with the Camper code and variables defined in background. Although I believe test runner would handle having the same name defined twice, in the Camper code and background, without errors, in one of previously changed challenges it was suggested to still avoid using the same name twice.

Reply:
It would clash, unfortunately.  You can test this by using
```
const testTriangle = [[6, 0, 0, 1],
                      [7, 4, 0, 0],
                      [2, 4, 6, 0],
                      [8, 5, 9, 3]];
```
in the challenge right now in production.  Even if the algorithm is correct, the tests will fail.

The test runner will not throw errors in this case, it will just silently replace any variables defined in `--before-user-code--` with the user's definitions.

Reply:
In production that would fail, because currently `testTriangle` is defined only in seed code :stuck_out_tongue:. 

Either way, despite of no explicit errors caused by repeated names, I get there's doze of surprising behavior with it, so it's best to avoid it all together.

Reply:
Ooof, good point.  What I was saying applies to `numTriangle`, at least. You can replace it the user code and break the tests.